daemon/containerio/cio.go

```diff
@@ -116,6 +116,12 @@ func copyIO(fifos *containerdio.FIFOSet, ioset *ioSet, tty bool) (_ *wgCloser, e
 		}
 	}()
 
+	// if fifos directory is not exist, create fifo will fails,
+	// also in case of fifo directory lost in container recovery process.
+	if _, err := os.Stat(fifos.Dir); err != nil && os.IsNotExist(err) {
```

Comment:
if the error is no the exist, should we return it directly?

Reply:
At first , 404 cause by http, then, we deal with non-exist file error to make fifo can be created successful.

Reply:
No, I means we only need to deal this kind error, and not check Mkdir's return.

Reply:
But the `Mkdir` maybe fail, right? And if `os.IsNotExist(err)` is false, I don't think it's necessary to execute the following statements. how do you think? 

cc @allencloud

Reply:
We do not  care about if mkdir' return, since if it fails , it means other problem, here we just deal with the problem the directory is not exist, so we also not deal with `os.IsNotExist(err)`.